packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,151 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SchemaFactory } from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+	export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+		node: [() => DeepTreeNode, schemaFactory.number],
+	}) {}
+	{
+		type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
+	}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export const WideTreeNode = schemaFactory.array(schemaFactory.number);
+export type WideTreeNode = NodeFromSchema<typeof WideTreeNode>;
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
```

Comment:
This doesn't really matter much since its just test code, but in case you didn't realize these issues, I thought I'd mention them since it can matter in production use cases.

The way this is implemented, if someone passed 1.5 into this (or any non-integer greater than 1, it will fail to terminate and stack overflow.

It also fails if someone used an integer large enough that subtracting 1 from it has no effect, Nan, or infinity.

The error message is also wrong if someone passes in 0.5 (it claims thats not greater than 0.)

Since this is test code, importing the node assert library and using it would also make sense.

The way this is implemented, if someone passed 1.5 into this (or any non-integer greater than 1, it will fail to terminate and stack overflow.

Also if someone used an integer large enough, subtracting 1 from it has no effect.



```suggestion
	assert(Number.isSafeInteger(depth));
	assert(depth > 0);
```

Reply:
Updated! Good catchup.